src/libraries/System.Private.CoreLib/src/System/Buffers/Binary/BinaryPrimitives.ReverseEndianness.cs

```diff
@@ -146,36 +145,38 @@ public static partial class BinaryPrimitives
                 // to end of the source and not have to worry about writing into the destination and clobbering
                 // source data we haven't yet read.
 
-                if (Vector128.IsHardwareAccelerated && source.Length >= Vector128<T>.Count)
+                int i = 0;
+
+                if (Vector256.IsHardwareAccelerated)
                 {
-                    if (Vector256.IsHardwareAccelerated && source.Length >= Vector256<T>.Count)
+                    while (i <= source.Length - Vector256<T>.Count)
                     {
-                        ref T end256 = ref Unsafe.Add(ref sourceRef, source.Length - Vector256<T>.Count + 1);
-                        do
-                        {
-                            Vector256.StoreUnsafe(TReverser.Reverse(Vector256.LoadUnsafe(ref sourceRef)), ref destRef);
-                            sourceRef = ref Unsafe.Add(ref sourceRef, Vector256<T>.Count);
-                            destRef = ref Unsafe.Add(ref destRef, Vector256<T>.Count);
-                        }
-                        while (Unsafe.IsAddressLessThan(ref sourceRef, ref end256));
+                        Vector256.StoreUnsafe(TReverser.Reverse(Vector256.LoadUnsafe(ref sourceRef, (nuint)i)), ref destRef, (nuint)i);
```

Comment:
```suggestion
                        Vector256.StoreUnsafe(TReverser.Reverse(Vector256.LoadUnsafe(ref sourceRef, (uint)i)), ref destRef, (uint)i);
```
The cast from int -> nuint is sign-extending. With the uint-cast it's just a `mov`.

Same below.